python/ray/rllib/models/action_dist.py

```diff
@@ -8,108 +8,108 @@
 
 
 class ActionDistribution(object):
-    """The policy action distribution of an agent.
+  """The policy action distribution of an agent.
 
-    Args:
-      inputs (Tensor): The input vector to compute samples from.
-    """
+  Args:
+    inputs (Tensor): The input vector to compute samples from.
+  """
 
-    def __init__(self, inputs):
-        self.inputs = inputs
+  def __init__(self, inputs):
+    self.inputs = inputs
 
-    def logp(self, x):
-        """The log-likelihood of the action distribution."""
-        raise NotImplementedError
+  def logp(self, x):
+    """The log-likelihood of the action distribution."""
+    raise NotImplementedError
 
-    def kl(self, other):
-        """The KL-divergence between two action distributions."""
-        raise NotImplementedError
+  def kl(self, other):
+    """The KL-divergence between two action distributions."""
+    raise NotImplementedError
 
-    def entropy(self):
-        """The entroy of the action distribution."""
-        raise NotImplementedError
+  def entropy(self):
+    """The entroy of the action distribution."""
+    raise NotImplementedError
 
-    def sample(self):
-        """Draw a sample from the action distribution."""
-        raise NotImplementedError
+  def sample(self):
+    """Draw a sample from the action distribution."""
+    raise NotImplementedError
 
 
 class Categorical(ActionDistribution):
-    """Categorical distribution for discrete action spaces."""
-
-    def logp(self, x):
-        return -tf.nn.sparse_softmax_cross_entropy_with_logits(
-            logits=self.inputs, labels=x)
-
-    def entropy(self):
-        a0 = self.inputs - tf.reduce_max(self.inputs, reduction_indices=[1],
-                                         keep_dims=True)
-        ea0 = tf.exp(a0)
-        z0 = tf.reduce_sum(ea0, reduction_indices=[1], keep_dims=True)
-        p0 = ea0 / z0
-        return tf.reduce_sum(p0 * (tf.log(z0) - a0), reduction_indices=[1])
-
-    def kl(self, other):
-        a0 = self.inputs - tf.reduce_max(self.inputs, reduction_indices=[1],
-                                         keep_dims=True)
-        a1 = other.inputs - tf.reduce_max(other.inputs, reduction_indices=[1],
-                                          keep_dims=True)
-        ea0 = tf.exp(a0)
-        ea1 = tf.exp(a1)
-        z0 = tf.reduce_sum(ea0, reduction_indices=[1], keep_dims=True)
-        z1 = tf.reduce_sum(ea1, reduction_indices=[1], keep_dims=True)
-        p0 = ea0 / z0
-        return tf.reduce_sum(p0 * (a0 - tf.log(z0) - a1 + tf.log(z1)),
-                             reduction_indices=[1])
-
-    def sample(self):
-        return tf.multinomial(self.inputs, 1)[0]
+  """Categorical distribution for discrete action spaces."""
```

Comment:
hey @eugenevinitsky can we keep the 4 spaces for formatting? Thanks!

Reply:
Oh shoot, yes!

Reply:
Ignore above, I'm not sure why it's showing up without four spaces

Reply:
@richardliaw I think that should do it?